raiden-ts/src/path/utils.ts

```diff
@@ -33,3 +44,105 @@ export function channelCanRoute(
     return `path: channel with "${partner}" don't have enough capacity=${capacity.toString()}`;
   return true;
 }
+
+const serviceRegistryToken = memoize(
+  async (serviceRegistryContract: ServiceRegistry) =>
+    serviceRegistryContract.functions.token() as Promise<Address>,
+);
+
+/**
+ * Returns a cold observable which fetch PFS info & validate for a given server address or URL
+ *
+ * @param pfsAddrOrUrl - PFS account/address or URL
+ * @param deps - RaidenEpicDeps needed for various parameters
+ * @returns Observable containing PFS server info
+ */
+export function pfsInfo(
+  pfsAddrOrUrl: Address | string,
+  { serviceRegistryContract, network, contractsInfo, config$ }: RaidenEpicDeps,
+): Observable<PFS> {
+  /**
+   * Codec for PFS /api/v1/info result schema
+   */
+  const PathInfo = t.type({
+    message: t.string,
+    network_info: t.type({
+      // literals will fail if trying to decode anything different from these constants
+      chain_id: t.literal(network.chainId),
+      registry_address: t.literal(contractsInfo.TokenNetworkRegistry.address),
+    }),
+    operator: t.string,
+    payment_address: Address,
+    price_info: UInt(32),
+    version: t.string,
+  });
+  // if it's an address, fetch url from ServiceRegistry, else it's already the URL
+  const url$ = Address.is(pfsAddrOrUrl)
+    ? from(serviceRegistryContract.functions.urls(pfsAddrOrUrl))
+    : of(pfsAddrOrUrl);
+  return url$.pipe(
+    withLatestFrom(config$),
+    mergeMap(([url, { httpTimeout }]) => {
+      if (!url || !url.includes('://')) throw new Error(`Invalid URL: ${url}`);
```

Comment:
Do we need to check for `https` as well or can we expect to only receive https URLs from the contract?

Reply:
@Dominik1999 could you help us here?

Reply:
I'd love to help. Can you elaborate on the question a bit more? Then either me or Karl can answer that, I am sure

Reply:
The question is whether PFS servers are always running on HTTPS and the ServiceRegistry always returns HTTPS URLs or if there is any requirement to support HTTP as well

Reply:
We could either require `startsWith('https://')` or add it on our own, seems there's at least one case on the current contracts where only the domain is set (without schema), which is filtered now, but could be useful to be kept instead of filtered out here.

Reply:
Done in #497